### ISSUE TITLE
(Self-Voicing) Help text spoken during interaction, not just on focus

### ISSUE THREAD

**Issue by terracoda** [phetsims/john-travoltage#394]
I was just playing with self-voicing for his sim and noticed a bug when "helpful hints" are enabled the interaction hint to move the leg plays on focus and after interaction with both the arm swing and the leg swing.

For this sim, this hint I am not sure if this hint should play at all outside the use of the Read-Me hint button. 

We can consider later what we do with that helpful hints checkbox if not help text exists in a sim, or if we need help text for the arm swing and leg swing, but for now though, 

@jessegreenberg, is it possible to just turn that hint off entirely when sim is in use? Have it play only with the read me button?

**Comment by terracoda** [comment]
Tagging @Matthew-Moore240, and @BLFiedler since there is an interview today on this sim.

**Comment by jessegreenberg** [comment]
@terracoda definitely possible to do so! This is not  a bug, just how I interpreted the current design doc, the "Interaction Hint" under the leg is listed as 
> Interaction hint: Swing John's leg to rub his foot on the rug.

So I thought it was associated with his leg rather than the button in the toolbar.

However, I just checked gravity-force-lab-basics and it looks like the behavior is that the hint is read every mouse click *if the object does not move*. Is that how it should behave in JT as well?

**Comment by terracoda** [comment]
Yes, I should have mentioned that the GFLB works differently. The helptext or "helpful hints" only read on focus or on click.

The delivery of the hint for mouse and touch might need to be restricted to a click or touch that does not result in a value change. 

A value change means they don't need any help :-)

Can you make JT's hint work the same as GFLB and then we can review.

**Comment by terracoda** [comment]
Back to you @jessegreenberg.

**Comment by jessegreenberg** [comment]
OK I updated the link. https://phet-dev.colorado.edu/html/jg-tests/john-travoltage_en_phet.html?supportsSelfVoicing

@terracoda because of the behavior we decided on in https://github.com/phetsims/john-travoltage/issues/377, the first grab response for the arm is **always** heard in full. So when dragging the arm with "Interaction Hints" enabled you might hear

"Arm swing, hand pointing straight up. Swing John's leg to rub his foot on the rug. Arm swing, hand close to doorknob."

This is an improvement, so I went with it for tomorrow. But still a little clunky. In order to fix this I think we should reconsider the behavior we decided in #377. If this is working for tomorrow @terracoda I think we should close this and open a new issue to reconsider #377.

**Comment by terracoda** [comment]
I think we need to re-visit the design choices in that were taken in #377 
There's too much repetition and it's slowing the experience. I will re-open #377 to review.

**Comment by terracoda** [comment]
@Matthew-Moore240 discussed some addition to help text for John Travoltage yesterday. I asked him to open an issue not remembering I had already opened one.

We discussed 2 changes:
1. Brief specific help text for the both the arm and leg to delivered if "Helpful hints" is checked.
  - Arm Swing: Move John's hand.
  - Leg Swing: Rub John's foot.
2. An edit to an existing phrase and an additional phrase for the dynamic interaction hint delivered through the read-me buttons.
  - **Change:** "Swing John's leg to rub his foot on rug." (remove "the" before "rug". "The rug" is mentioned in the overview, so no need for "the rug" in the hint - "rug" is good on its own.)
  - **No change:** "Move John's hand to see when he gets a shock."
  - **New dynamic hint:** "Rub John's foot some more
      - Logic for new hint: If John's body has electrons, but not enough to give him a shock (i.e. less than "several").

**Comment by terracoda** [comment]
I think we agreed on changes in https://github.com/phetsims/john-travoltage/issues/394#issuecomment-800209903 March 16th Self-Voicing meeting.

**Comment by terracoda** [comment]
When the "Helpful hint" (i.e. help text for individual interactions) is spoken is described over in #377.

**Comment by jessegreenberg** [comment]
@terracoda @Matthew-Moore240 here is a new version with changes listed above.

https://phet-dev.colorado.edu/html/jg-tests/john-travoltage_en_phet.html?supportsSelfVoicing

Can you please review?

**Comment by terracoda** [comment]
@jessegreenberg, this sounds really great!

For the Leg Swing, I noticed only get the Leg Swing position on focus, which I though was odd at first, but since I get the electron response every rub, it sounds quite nice. However, the foot is really far off the rug, I get no voicing, so we might need to put foot position.

Was it intentional to leave off the Leg Swing position during changes?

Regardless, I'll use this version for tomorrow. It sounds good to me.

**Comment by terracoda** [comment]
And the help text and dynamic read-me Hint all sound good.

**Comment by terracoda** [comment]
I'll close this issue move review of Leg Swing position information to a new issue.
